src/test/groovy/graphql/execution/instrumentation/dataloader/DataLoaderTypeMismatchTest.groovy

```diff
@@ -0,0 +1,74 @@
+package graphql.execution.instrumentation.dataloader
+
+import graphql.GraphQL
+import graphql.schema.DataFetcher
+import graphql.schema.DataFetchingEnvironment
+import graphql.schema.idl.RuntimeWiring
+import graphql.schema.idl.SchemaGenerator
+import graphql.schema.idl.SchemaParser
+import org.dataloader.BatchLoader
+import org.dataloader.DataLoader
+import org.dataloader.DataLoaderRegistry
+import spock.lang.Specification
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+
+import static graphql.schema.idl.TypeRuntimeWiring.newTypeWiring
+
+class DataLoaderTypeMismatchTest extends Specification {
+
+    def "when actual field value return type is different from expected return type, then it should not hang execution"() {
+        setup:
+        def sdl = """
+        type Todo {
+           id: ID!
+        }
+
+        type Query {
+           getTodos: [Todo]
+        }
+
+        schema {
+           query: Query
+        }"""
+
+        def typeDefinitionRegistry = new SchemaParser().parse(sdl)
```

Comment:
Checkout https://github.com/graphql-java/graphql-java/blob/master/src/test/groovy/graphql/TestUtil.groovy for a testing helper that does parsing and schema generation in tests.  Mostly for future reference

Reply:
Ok sounds good, thanks for the tip